### PR TITLE
Task akazakov add virtualization type support SIO-3711

### PR DESCRIPTION
**Pull request by cyxrop**

### CODE REVIEW

File: osImageVersions.go [hunk 1]
```diff
@@ -3,60 +3,62 @@ package solus
...
+	Version            string                 `json:"version"`
+	URL                string                 `json:"url"`
+	CloudInitVersion   null.String            `json:"cloud_init_version"`
+	OsImageID          int                    `json:"os_image_id"`
+	IsVisible          bool                   `json:"is_visible"`
+	IsSSHKeysSupported bool                   `json:"is_ssh_keys_supported"`
 }
 
-// CloudInitVersion represents available cloud-init config versions.
-type CloudInitVersion string
```
Comment (dshemin) [inline osImageVersions.go:24]: Can we use `type CloudInitVersion null.String`?

File: osImageVersions.go [hunk 2]
```diff
@@ -3,60 +3,62 @@ package solus
...
-	CloudInitVersion   CloudInitVersion `json:"cloud_init_version"`
-	OsImageID          int              `json:"os_image_id"`
-	IsVisible          bool             `json:"is_visible"`
-	IsSSHKeysSupported bool             `json:"is_ssh_keys_supported"`
+	ID                 int                    `json:"id"`
+	VirtualizationType VirtualizationTypeName `json:"virtualization_type"`
+	Position           float64                `json:"position"`
+	Version            string                 `json:"version"`
+	URL                string                 `json:"url"`
+	CloudInitVersion   null.String            `json:"cloud_init_version"`
```
Comment (dshemin) [inline osImageVersions.go:19]: Can we use just use an ordinal string with `omitempty` instead?

File: osImageVersions_test.go [hunk 3]
```diff
@@ -10,40 +10,52 @@ import (
...
-		Position:         1.5,
-		Version:          "version",
-		URL:              "http://foo/bar",
-		CloudInitVersion: CloudInitVersionV0,
-		IsVisible:        true,
+		Position:           1.5,
+		Version:            "version",
+		VirtualizationType: VirtualizationTypeKVM,
+		URL:                "http://foo/bar",
+		CloudInitVersion:   CloudInitVersionV0,
```
Comment (dshemin) [inline osImageVersions_test.go:43]: Can we use just use an ordinal string with `omitempty` instead?

File: computeResource.go [hunk 4]
```diff
@@ -6,6 +6,17 @@ import (
 	"net/http"
 )
 
+// VirtualizationTypeName represents available storage types.
```
Comment (dshemin) [inline computeResource.go:9]: Not "storage types" but virtualization types.

Comment (cyxrop) [reply computeResource.go]: done

**Pull request merged**